marimbabot_behavior/src/marimbabot_behavior/behavior_node.py

```diff
@@ -151,69 +225,110 @@ def callback_command(self, command_msg):
                 rospy.logwarn('No notes recognized.')
                 self.response_pub.publish('No notes recognized.')
 
-        # play the notes on the marimba using the UR5
-        elif command == 'marimbabot start playing':
+        # play the notes on the marimba using the UR5 or generate an audio preview
+        elif re.match(r'marimbabot start (playing|preview) (in a loop)?', command):
             # if a note sequence has been read via the 'read' command and the corresponding hit sequence is valid, the hit sequence is send to the planning action server
             if self.hit_sequence:
-                self.play()
+                # check if the sequence should be played in a loop
+                loop = True if 'loop' in command else False
+                # create event to stop the play if the 'stop' command is issued
+                self.event = threading.Event()
+
+                # if the motion is not busy, the hit sequence is send to the planning action server 
+                if 'playing' in command:
+                    if self.planning_client.simple_state == actionlib.SimpleGoalState.DONE:
+                        self.play(loop)
+                    else:
+                        rospy.logwarn('The robot is already playing.')
+                        self.response_pub.publish('The robot is already playing.')
+
+                # if the preview is not busy, the note sequence is send to the lilypond_audio action server
+                if 'preview' in command:
+                    if self.lilypond_audio_client.simple_state == actionlib.SimpleGoalState.DONE:
+                        self.preview(loop)
+                    else:
+                        rospy.logwarn('The audio preview is already playing.')
+                        self.response_pub.publish('The audio preview is already playing.')
             else:
                 rospy.logwarn('No notes to play. Say reed to read notes.')
                 self.response_pub.publish('No notes to play. Say reed to read notes.')
```

Comment:
typo `reed`

Reply:
I fixed the type in the logwarn message. Unfortunately, it has to stay in the response string, as the TTS would pronounce it as past tense otherwise (rĕd instead of rēd)